elastic-search-support/src/main/java/uk/gov/hmcts/ccd/definition/store/elastic/client/HighLevelCCDElasticClient.java

```diff
@@ -106,4 +115,99 @@ private String getCurrentAliasIndex(String indexName, GetAliasesResponse aliases
+    }
+
+    public boolean createIndexAndMapping(String indexName, String caseTypeMapping) throws IOException {
+        //create new index
+        CreateIndexRequest request = new CreateIndexRequest(indexName);
+        request.settings(casesIndexSettings(CASES_INDEX_SETTINGS_JSON));
+        CreateIndexResponse createIndexResponse = elasticClient.indices().create(request, RequestOptions.DEFAULT);
+        log.info("index created: {}", createIndexResponse.isAcknowledged());
+
+        //upsert mapping to new index
+        PutMappingRequest putRequest = new PutMappingRequest(indexName);
+        putRequest.source(caseTypeMapping, XContentType.JSON);
+
+        AcknowledgedResponse acknowledgedResponse = elasticClient.indices()
+            .putMapping(putRequest, RequestOptions.DEFAULT);
+        log.info("mapping upserted: {}", acknowledgedResponse.isAcknowledged());
+
+        return createIndexResponse.isAcknowledged();
+    }
+
+    public boolean updateAlias(String aliasName, String oldIndex, String newIndex) throws IOException {
+        IndicesAliasesRequest aliasRequest = new IndicesAliasesRequest();
+        aliasRequest.addAliasAction(IndicesAliasesRequest.AliasActions.remove().index(oldIndex).alias(aliasName));
+        aliasRequest.addAliasAction(IndicesAliasesRequest.AliasActions.add().index(newIndex).alias(aliasName));
+
+        AcknowledgedResponse aliasResponse = elasticClient.indices()
+            .updateAliases(aliasRequest, RequestOptions.DEFAULT);
+        if (aliasResponse.isAcknowledged()) {
+            log.info("alias successfully updated: {} now points to {}", oldIndex, newIndex);
+        } else {
+            log.info("alias update failed: {} still points to {}", oldIndex, oldIndex);
+        }
+        return aliasResponse.isAcknowledged();
+    }
+
+    public boolean removeIndex(String indexName) throws IOException {
+        DeleteIndexRequest deleteIndexRequest = new DeleteIndexRequest(indexName);
+        AcknowledgedResponse deleteResponse = elasticClient.indices()
+            .delete(deleteIndexRequest, RequestOptions.DEFAULT);
+        if (deleteResponse.isAcknowledged()) {
+            log.info("successfully deleted index: {}", indexName);
+        } else {
+            log.info("failed to delete index: {}", indexName);
+        }
+        return deleteResponse.isAcknowledged();
+    }
+
+    public CompletableFuture<String> reindexData(String oldIndex, String newIndex) {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        try {
+            String jsonBody = String.format(
+                "{ \"source\": { \"index\": \"%s\" }, \"dest\": { \"index\": \"%s\" } }",
+                oldIndex, newIndex
+            );
+
+            // send reindex request with wait_for_completion=false for asynchronus reindexing to get task id
+            Request request = new Request("POST", "_reindex?wait_for_completion=false");
+            request.setJsonEntity(jsonBody);
```

Comment:
Try this instead of above code:
ReindexRequest reindexRequest = new ReindexRequest(); 
reindexRequest.setSourceIndices(oldIndex);
reindexRequest.setDestIndex(newIndex);
reindexRequest.setRefresh(true);

TaskSubmissionResponse reindexSubmission = highLevelClient()
    .submitReindexTask(reindexRequest, RequestOptions.DEFAULT); 

String taskId = reindexSubmission.getTask();